### PR TITLE
Make --emit dep-info work correctly with -Z no-analysis again.

### LINKED ISSUE #33231: dep-info + -Z no-analysis no longer ignore extern crates
Compiling this code:

```rust
extern crate doesnt_exist;

fn main() {}
```

With:

```
rustc -Z no-analysis --emit dep-info -o test.d test.rs
```

Used to work (and still works with rust 1.7), but recently has started failing (circa rustc 1.10.0-nightly (bd938166d 2016-04-25)) with this error:

```
test.rs:1:1: 1:27 error: can't find crate for `doesnt_exist` [E0463]
test.rs:1 extern crate doesnt_exist;
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
error: aborting due to previous error
```

This makes that useful combination of flags no longer work. I suspect `-Z no-analysis` broke again.

### PR DESCRIPTION
**Pull request by SiegeLord**
Previously, it would attempt to resolve some external crates that weren't necessary for dep-info output.

Fixes #33231.

### CODE REVIEW

File: src/librustc_driver/driver.rs [hunk 1]
```diff
@@ -161,23 +174,20 @@ pub fn compile_input(sess: &Session,
                            "indexing hir",
                            move || hir_map::map_crate(hir_forest, defs));
 
-
-        write_out_deps(sess, &outputs, &id);
-
         {
             let _ignore = hir_map.dep_graph.in_ignore();
-            controller_entry_point!(after_write_deps,
+            controller_entry_point!(after_ast,
```
Comment (nrc) [inline src/librustc_driver/driver.rs]: I think `after_hir_lowering` might be a better name here - `ast` is a data structure rather than a phase.

**Comment by nrc** [comment]
lgtm, r+ with the name changed

File: src/librustc_driver/driver.rs (hunk 1 above)
Comment (SiegeLord) [inline src/librustc_driver/driver.rs]: Done.

**Comment by SiegeLord** [comment]
Whoops, looks like my regex failed on Windows. Could you approve again?

**Comment by SiegeLord** [comment]
Re-based onto master.

**Comment by oli-obk** [comment]
This failed the rollup:

```
---- [run-make] run-make/dep-info-no-analysis stdout ----
	
error: make failed
status: exit code: 2
command: "make"
stdout:
------------------------------------------
DYLD_LIBRARY_PATH="/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/test/run-make/dep-info-no-analysis.stage2-x86_64-apple-darwin:/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/stage2/lib:/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/stage2-tools/x86_64-apple-darwin/release/deps:/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/stage2-rustc/x86_64-apple-darwin/release/deps:/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/stage2-test/x86_64-apple-darwin/release/deps:/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/stage2-std/x86_64-apple-darwin/release/deps:" '/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/stage2/bin/rustc' --out-dir /Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/test/run-make/dep-info-no-analysis.stage2-x86_64-apple-darwin -L /Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/test/run-make/dep-info-no-analysis.stage2-x86_64-apple-darwin  -o /Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/test/run-make/dep-info-no-analysis.stage2-x86_64-apple-darwin/input.dd -Z no-analysis --emit dep-info input.rs
sed -i "s/^.*input.dd/input.dd/g" /Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/obj/build/x86_64-apple-darwin/test/run-make/dep-info-no-analysis.stage2-x86_64-apple-darwin/input.dd

------------------------------------------
stderr:
------------------------------------------
warning: ignoring --out-dir flag due to -o flag.
sed: -i may not be used with stdin
make[1]: *** [all] Error 1

------------------------------------------

thread '[run-make] run-make/dep-info-no-analysis' panicked at 'explicit panic', /Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/src/tools/compiletest/src/runtest.rs:1543
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    [run-make] run-make/dep-info-no-analysis
```

**Comment by SiegeLord** [comment]
::sigh:: this wouldn't be a problem if rust didn't put the absolute path inside the deps file, which requires me to sed it away...

**Comment by SiegeLord** [comment]
I fixed the sed invocation to work with BSD utils, so should be ok to try this again.

**Pull request merged**